### ISSUE TITLE
Publish to Github Not Functional?

### ISSUE THREAD

**Issue by quadrophobiac** [theodi/comma-chameleon#178]
[PLACEHOLDER, will be edited more comprehensively, relates to https://github.com/theodi/comma-chameleon/pull/177]
> Please provide a general summary of the issue in the Issue Title above
> fill out the headings below as applicable to the issue you are reporting,
> deleting as appropriate but offering us as much detail as you can to help us resolve the issue

### Expected Behaviour
> What should happen?

### Desired Behaviour (for enhancement suggestions only)
> if relevant include images or hyperlinks to other resources that clarify the enhancement you're seeking

### Current Behaviour (for problems)
> What currently happens that isn't expected behaviour?

### Steps to Reproduce (for problems)
> Provide a link to a live example, or an unambiguous set of steps to reproduce this bug. Include code to reproduce, if relevant
1.
2.
3.
4.

### Your Environment
> Include as many relevant details about the environment you experienced the bug in - this will help us resolve the bug more expediently
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):

**Comment by quadrophobiac** [comment]
@pezholio realised you may not get a ping when I edit an existing comment so pinging anew here

**Comment by pezholio** [comment]
Huh. That's weird. I get a new window that prompts me to login to Github, but when I try to submit a file, I get `Dataset files is invalid`. Were any changes made to the Octopub API since the FSA work was done in the new year?

I haven't got time at the mo to dig into this, but if it helps, the relevant backend code that talks to the Octopub API is [here](https://github.com/theodi/comma-chameleon/blob/master/main/github.js)

Sidebar: When logging into Github to get my token, and trying to copy and paste my Github password from my password manager, I get this error.

```
Uncaught Exception:
ReferenceError: mainWindow is not defined
    at exports.menu.submenu.click (/Applications/Comma Chameleon.app/Contents/Resources/app/main/menu.js:187:11)
    at MenuItem.click (/Applications/Comma Chameleon.app/Contents/Resources/electron.asar/browser/api/menu-item.js:81:16)
    at Function.delegate.executeCommand (/Applications/Comma Chameleon.app/Contents/Resources/electron.asar/browser/api/menu.js:119:40)
```

**Comment by chris48s** [comment]
@pezholio - that issue should have been fixed in f21beb4

If you're using a binary dist, I think that commit hasn't made it into a tagged release yet.

If you're running from source, make sure you've got the latest commits fetched/pulled, although @quadrophobiac has been [having some problems](https://github.com/theodi/comma-chameleon/issues/164) with that so I'd be interested to know if you have the same issues.

**Comment by Floppy** [comment]
API shouldn't have changed, no. We should re-test with the current master and tag a release.

**Comment by quadrophobiac** [comment]
re-test Comma Chameleon you mean?

**Comment by Floppy** [comment]
Yep

**Comment by chris48s** [comment]
Yes - I've got the same problem. Tests pass, 'Export to GitHub' gives me a blank window with no errors logged to the console.

There's definitely something odd going on here though. Lets go back in time to a point where I know this feature definitely worked...

If I download the binary dist for say version 0.4.7, that feature does work .. but then on my local copy, if I `git checkout  af7f60e` ( af7f60e  is the commit tagged as 0.4.7), I've got the same problem on my working copy. I think whatever is going on is going to be related to something that is happening in the build process or a dependency or a setting that is on in dev but not when we build a release or something like that.. I don't think the actual "export to GitHub" code is fundamentally broken.

Sorry I don't have more time to spend on it right now but maybe that gives you a clue?
